src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -779,21 +767,20 @@ private fun processNestedScopesUpwards(scopeStart: RsCompositeElement, processor
     return false
 }
 
-
 // There's already similar functions in TreeUtils, should use it
 private fun walkUp(
-    start: RsCompositeElement,
+    start: PsiElement,
     stopAfter: (RsCompositeElement) -> Boolean,
-    processor: (cameFrom: RsCompositeElement, scope: RsCompositeElement) -> Boolean
+    context: (cameFrom: PsiElement, scope: RsCompositeElement) -> RsCompositeElement?,
```

Comment:
Hm, maybe we can add a `stopAtFileBoundary: Boolean = false` flag instead of making this fully configurable via a closure?